src/pages/BuySell/SelectLayout.js

```diff
@@ -0,0 +1,218 @@
+import { React, useState } from 'react';
+import { useNavigate } from 'react-router-dom';
+import styled from 'styled-components';
+import { flexBox } from '../../styles/mixin';
+
+const SIZE_BTN = [
+  { size: 230 },
+  { size: 240 },
+  { size: 250 },
+  { size: 260 },
+  { size: 270 },
+  { size: 280 },
+];
+
+const SelectLayout = ({ type }) => {
+  const [selectSize, setSelectSize] = useState();
+
+  const navigate = useNavigate();
+
+  const onSizeClick = size => setSelectSize(size);
+
+  return (
+    <BuyBackGround>
+      <Title>{type === 'sell' ? '판매하기' : '구매하기'}</Title>
+      <BuyWrapper>
+        <BuyBox>
+          <ItemInfoBox>
+            <ItemImg
+              src="http://placeimg.com/640/640/people"
+              alt="제품 이미지"
+              width="80px"
+            />
+            <ItemInfo>
+              <ItemCode>DZ5485-612</ItemCode>
+              <ItemEnglishName>
+                Jordan 1 Retro High OG Chicago 2022
+              </ItemEnglishName>
+              <ItemKoreanName>조던 1 레트로 하이 OG 시카고 2022</ItemKoreanName>
+            </ItemInfo>
+          </ItemInfoBox>
+
+          <ItemSelectBox>
+            <SelectList>
+              {SIZE_BTN.map(({ size }) => {
+                return (
+                  <SelectItem
+                    onClick={() => onSizeClick(size)}
+                    className={size === selectSize && 'active'}
+                    key={size}
+                  >
+                    <SizeBtn>
+                      <Size>{size}</Size>
+                      <Price type={type}>123,000</Price>
+                    </SizeBtn>
+                  </SelectItem>
+                );
+              })}
+            </SelectList>
+          </ItemSelectBox>
+          {selectSize && (
+            <BtnArea>
+              <BtnBox
+                onClick={() => {
+                  navigate(`/${type}`);
+                }}
```

Comment:
navigate 되었을 시 size를 전달 할 수 없는 것 같은데 size는 어떻게 핸들링 하시나요?
사실 state로 관리되어있는게 가장 좋지만, navigate시에 비하인더씬으로 state를 넘기는 방법 중에 아래와 같은 방식도 있습니다. 

```suggestion
                onClick={() => {
                  navigate(`/${type}`, { state: { size: selectSize } });
                }}
```

사용하는 측에서 콘솔 찍어보시고 필요한 state 활용하시면 좋을 것 같아요! 
```js
const location = useLocation();
console.log(location) 
```